src/current.rs

```diff
@@ -66,20 +68,26 @@ impl CURRENT_TASK_PTR_WRAPPER {
             let base: usize;
             #[cfg(target_arch = "x86_64")]
             {
-                core::arch::asm!(
-                    "mov {0}, gs:[offset __PERCPU_SELF_PTR]",
-                    "add {0}, offset {VAR}",
-                    out(reg) base,
-                    VAR = sym __PERCPU_CURRENT_TASK_PTR,
-                );
+                unsafe {
```

Comment:
Can you explain the reason to add `unsafe` block here when the functions have been marked unsafe? Or you can paste the warning or error reported by the compiler.Thanks.

Reply:
It's the suggstion from unsafe_op_in_unsafe_fn lint, as stated in PR description

> mostly around adding unsafe block for calling unsafe code (refer to [unsafe_op_in_unsafe_fn](https://doc.rust-lang.org/stable/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html) for more info)

A diagnostic example on [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=3236e7b85572df65ee04a3bfdb3124e3):

```rust
warning[E0133]: call to unsafe function `core::slice::<impl [T]>::get_unchecked` is unsafe and requires unsafe block
 --> src/lib.rs:2:3
  |
2 |   x.get_unchecked(i) // WARNING: requires unsafe block
  |   ^^^^^^^^^^^^^^^^^^ call to unsafe function
  |
  = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html>
  = note: consult the function's documentation for information on how to avoid undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
 --> src/lib.rs:1:1
  |
1 | pub unsafe fn get_unchecked<T>(x: &[T], i: usize) -> &T {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: `#[warn(unsafe_op_in_unsafe_fn)]` on by default

For more information about this error, try `rustc --explain E0133`.
```


---

I'll quote from the linked edition guide book:

> This change is intended to help protect against accidental use of unsafe operations in an unsafe function. The unsafe function keyword was performing two roles. One was to declare that calling the function requires unsafe, and that the caller is responsible to uphold additional safety requirements. The other role was to allow the use of unsafe operations inside of the function. This second role was determined to be too risky without explicit unsafe blocks.
>
>More information and motivation may be found in [RFC #2585](https://rust-lang.github.io/rfcs/2585-unsafe-block-in-unsafe-fn.html).

Reply:
Ok, thanks